tests/unit/test_coloring.py

```diff
@@ -0,0 +1,200 @@
+"""
+Test file for Coloring class
+"""
+import unittest
+from coloring import *
+from graph_io import load_graph
+
+
+class ColoringCase(unittest.TestCase):
+    coloring = None
+    graph = None
+    v1 = None
+    v2 = None
+
+    @classmethod
+    def create_coloring_helper(self, vertices: List[int], map: dict):
+        coloring = Coloring()
+        for color in map:
+            for value in map[color]:
+                vertex = [v for v in vertices if v.label == value][0]
+                coloring.set(vertex, color)
+        return coloring
+
+    @classmethod
+    def create_graph_helper(self, edges: List[List[int]]):
+        g = Graph(False)
+        vertices = {}
+        for edge in edges:
+            head, tail = edge
+            if head not in vertices:
+                vertices[head] = Vertex(g, label=head)
+                g.add_vertex(vertices[head])
+            if tail not in vertices:
+                vertices[tail] = Vertex(g, label=tail)
+                g.add_vertex(vertices[tail])
+            g.add_edge(Edge(vertices[head], vertices[tail]))
+        return g
+
+    def setUp(self):
+        self.coloring = Coloring()
+        self.graph = Graph(False, 5)
+        self.v1, self.v2 = Vertex(self.graph), Vertex(self.graph)
+
+    def test_init(self):
+        # Has no colors or vertices
+        self.assertEqual(0, len(self.coloring))
+        self.assertEqual(0, len(self.coloring.vertices))
+
+    def test_set(self):
+        # Adds a new color if not existing
+        self.coloring.set(self.v1, 4)
+        self.assertEqual(1, len(self.coloring))
+        self.assertEqual(4, list(self.coloring.colors)[0])
+
+        # Adds no new color if already existing
+        self.coloring.set(self.v2, 4)
+        self.assertEqual(1, len(self.coloring))
+
+        # Cannot set a vertex that is already in the coloring (have to recolor it)
+        with self.assertRaises(KeyError) as e:
+            self.coloring.set(self.v1, 0)
+        self.assertEqual('\'Vertex {} already in coloring, color: 4\''.format(str(self.v1)), str(e.exception))
+
+    def test_get(self):
+        self.coloring.set(self.v1, 4)
+        self.coloring.set(self.v2, 4)
+        vertices = self.coloring.get(4)
+        self.assertEqual(2, len(vertices))
+        self.assertEqual(True, self.v1 in vertices)
```

Comment:
Je kan ook `self.assertTrue` gebruiken

Reply:
Geldt op meerdere plekken waar je `self.assertEqual(True/False, ...)` hebt gedaan

Reply:
De foutmessage van een assertTrue/assertFalse is meestal erg minimaal. Vandaar dat ik het zo doe.